api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -1,173 +1,60 @@
-  -- structured data composed of arrays and maps of other values. Body MUST support any type to preserve the semantics of structured logs emitted by the applications.
-  -- Can vary for each occurrence of the event coming from the same source. This field is optional.
-  --
-  -- Type any (functions that use Log should have a Typeclass constraint of (ToValue body) => ...)
-  --    Value of type any can be one of the following:
-  --    - A scalar value: number, string or boolean,
-  --    - A byte array,
-  --    - An array (a list) of any values,
-  --    - A map<string, any>.
-  , resource :: Maybe MaterializedResources
-  -- ^ Describes the source of the log, aka resource. Multiple occurrences of events coming from the same event source can happen across time and they all have the same value of Resource.
-  -- Can contain for example information about the application that emits the record or about the infrastructure where the application runs. Data formats that represent this data model
-  -- may be designed in a manner that allows the Resource field to be recorded only once per batch of log records that come from the same source. SHOULD follow OpenTelemetry semantic conventions for Resources.
-  -- This field is optional.
-  , attributes :: LogAttributes
-  -- ^ Additional information about the specific event occurrence. Unlike the Resource field, which is fixed for a particular source, Attributes can vary for each occurrence of the event coming from the same source.
-  -- Can contain information about the request context (other than Trace Context Fields). The log attribute model MUST support any type, a superset of standard Attribute, to preserve the semantics of structured attributes
-  -- emitted by the applications. This field is optional.
-  }
-  deriving stock (Functor)
+getCurrentTimestamp :: (MonadIO m) => m Timestamp
+getCurrentTimestamp = liftIO $ coerce @(IO TimeSpec) @(IO Timestamp) $ getTime Realtime
+
 
-{- data SeverityNumber
-  = Trace
-  | Trace2
-  | Trace3
-  | Trace4
-  | Debug
-  | Debug2
-  | Debug3
-  | Debug4
-  | Info
-  | Info2
-  | Info3
-  | Info4
-  | Warn
-  | Warn2
-  | Warn3
-  | Warn4
-  | Error
-  | Error2
-  | Error3
-  | Error4
-  | Fatal
-  | Fatal2
-  | Fatal3
-  | Fatal4
-  | Unknown !Int32
-  deriving (Eq, Ord, Read, Show )
+{- | Emits a LogRecord with properties specified by the passed in Logger and LogRecordArguments.
+If observedTimestamp is not set in LogRecordArguments, it will default to the current timestamp.
+If context is not specified in LogRecordArguments it will default to the current context.
+-}
+emitLogRecord
+  :: (MonadIO m)
+  => Logger
+  -> LogRecordArguments body
+  -> m (LogRecord body)
+emitLogRecord logger LogRecordArguments {..} = do
+  currentTimestamp <- getCurrentTimestamp
+  let logRecordObservedTimestamp = fromMaybe currentTimestamp observedTimestamp
```

Comment:
Uses the supplied `observedTimestamp` if there is one, but otherwise uses the `currentTimestamp`.